### PR TITLE
3 pc model fallback

### LINKED ISSUE #5425: 3PC model fallback when the "b free" is not available
In the current implementation of the 3PC, the spectral model that one gets from a source is the "ECPL_bfr", meaning a `SuperExpCutoffPowerLaw4FGLDR3SpectralModel` with `index_2` free. However, there are some case (like PSR J0540-6919) where this fit didn't work, so the value in the 3PC table are set to 0. This means that we return a model with parameter set to 0 to the user. 
But most of the time the fit of the `SuperExpCutoffPowerLaw4FGLDR3SpectralModel` with `index_2` fixed at 2/3 worked and we can use it instead. 

In our code, we need to fallback to this "ECPL_b23" if "ECPL_bfr" is not available.

### PR DESCRIPTION
**Pull request by MRegeard**
This pull request fix #5425. 

I checked if the index of the spectral model is masked in the "b free" fit. if it is the case, I return the "b 2/3" fit.

### CODE REVIEW

**Review by QRemy — APPROVED** [review]
Thanks @MRegeard, no comment from my side.

File: gammapy/catalog/fermi.py [hunk 1]
```diff
@@ -1655,19 +1650,33 @@ def spectral_model(self):
...
+            pars = {
+                "reference": d["Pivot_Energy_b23"],
+                "amplitude": d["PLEC_Flux_Density_b23"],
+                "index_1": -d["PLEC_IndexS_b23"],
+                "index_2": d["PLEC_Exp_Index_b23"],
+                "expfactor": d["PLEC_ExpfactorS_b23"],
+            }
+            errs = {
+                "amplitude": d["Unc_PLEC_Flux_Density_b23"],
+                "index_1": d["Unc_PLEC_IndexS_b23"],
```
[the remark this answers is not in the document]
Comment (MRegeard) [reply gammapy/catalog/fermi.py:1677]: I think that the error is initialised at 0 if no value is given.

**Comment by MRegeard** [comment]
Thanks @AtreyeeS, I agree with you. What we could also do is add an argument to the `spectral_model` method that specify the model the user want. By default it would be "b free", but the user could pass "b 23" which would return the other model. What do you think?

File: gammapy/catalog/fermi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AtreyeeS) [reply gammapy/catalog/fermi.py:1677]: Okay. The you can assert it in the tests?

**Comment by AtreyeeS** [comment]
> By default it would be "b free", but the user could pass "b 23" which would return the other model. What do you think?

I dont have experience with the pulsar models, but in general I agree.
Instead of silently going to the fallback option, it will be better to know that the b free fit failed.

You can mention accordingly in the docstring

**Review by registerrier — APPROVED** [review]
Thanks @MRegeard . Looks good to me!

**Review by AtreyeeS — APPROVED** [review]
Thanks @MRegeard ! No more comments

**Pull request merged**